### PR TITLE
Remove moments centering and tests improvements

### LINKED ISSUE #133: Moments centering
Remove explicit support for centering from moments.
Remove checks like `isinstance(values, (int, float))`.

### LINKED ISSUE #134: pytest - create variables inside functions
Create variables inside functions, each run should have independent variables. see `test_sampling_pools.py`

### PR DESCRIPTION
**Pull request by martinspetlik**
resolve #133
resolve #134

### CODE REVIEW

**Comment by martinspetlik** [comment]
resolve #134

File: src/mlmc/quantity_spec.py [hunk 1]
```diff
@@ -10,8 +11,14 @@ class QuantitySpec:
     times: List[float]
     locations: Union[List[str], List[Tuple[float, float, float]]]
 
+    def __eq__(self, other):
```
Comment (jbrezmorf) [inline src/mlmc/quantity_spec.py]: The equality should be exact. Any other reason for own implementation.

File: src/mlmc/sample_storage_hdf.py [hunk 2]
```diff
@@ -86,6 +86,11 @@ def save_result_format(self, result_format: List[QuantitySpec], res_dtype):
         :param result_format: List[QuantitySpec]
         :return: None
         """
+        try:
```
Comment (jbrezmorf) [inline src/mlmc/sample_storage_hdf.py:89]: The result format must be set before storing the values, we should probably provide it in the constructor.

File: test/test_run.py [hunk 3]
```diff
@@ -45,27 +41,18 @@ def hdf_storage_factory():
...
-                                       (simulation, hdf_storage_factory(), sampling_pool_single_process),
-                                       (simulation, hdf_storage_factory(), sampling_pool_processes),
-                                       #(simulation, hdf_storage_factory(), sampling_pool_thread),
-                                       (simulation_workspace, storage_memory, sampling_pool_single_process_dir),
-                                       (simulation_workspace, storage_memory, sampling_pool_processes_dir),
-                                       #(simulation_workspace, storage_memory, sampling_pool_thread_dir),
-                                       (simulation_workspace, hdf_storage_factory(), sampling_pool_single_process_dir),
-                                       (simulation_workspace, hdf_storage_factory(), sampling_pool_processes_dir),
-                                       #(simulation_workspace, hdf_storage_factory(), sampling_pool_thread_dir)
+@pytest.mark.parametrize("test_case", [(SynthSimulationForTests(simulation_config), Memory(), OneProcessPool()),
```
Comment (jbrezmorf) [inline test/test_run.py:44]: To preserve clarity one can decalre a factory functions:
```
def sampling_pool_single_process():
   return OneProcessPool()
```
... only meaningful for more parameters.

**Pull request merged**